sidedata.go

```diff
@@ -72,51 +81,56 @@ type SideDataMasteringDisplayMetadata struct {
 
 // SideDataContentLightLevel represents the content light level side data.
 type SideDataContentLightLevel struct {
+	SideDataBase
 	MaxContent int `json:"max_content,omitempty"`
 	MaxAverage int `json:"max_average,omitempty"`
 }
 
 // SideDataUnknown represents an unknown side data.
-type SideDataUnknown []Tags
+type SideDataUnknown Tags
 
 // SideData represents a side data packet.
 type SideData struct {
-	SideDataType string `json:"side_data_type"`
-	Data         interface{}
+	SideDataBase
+	Data interface{} `json:"-"`
 }
 
-// Custom UnmarshalJSON to decode into specific structure based on SideDataType
 func (sd *SideData) UnmarshalJSON(b []byte) error {
-	var raw = &struct {
-		SideDataType string `json:"side_data_type"`
-	}{}
+	type Alias SideData
```

Comment:
Why is this type alias necessary? 🤔

Reply:
This is necessary to avoid recursion, as the source type has a Marshal method.